### PR TITLE
Fixed clang configuration file

### PR DESCRIPTION
**Pull request by Hadatko**
+ reformatted code.

# Pull request

**Choose Correct**

- [ ] bug
- [X] feature

**Describe the pull request**

I noticed that some clang-format config values were changed for version we have. So i adapt them and added one more which was not available.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

- OS:
- eRPC Version:

**Steps you didn't forgot to do**

- [X] I checked if other PR isn't solving this issue.
- [X] I read Contribution details and did appropriate actions.
- [X] PR code is tested.
- [X] PR code is formatted.
- [X] Allow edits from maintainers pull request option is set (recommended).

**Additional context**

### CODE REVIEW

**Comment by MichalPrincNXP** [comment]
Hi Dusan, thank you. Would it be possible to move to LLVM16 to be aligned with MCUXpressoSDK formatting?

**Review by MichalPrincNXP — CHANGES_REQUESTED** [review]

File: .clang-format [hunk 1]
```diff
@@ -45,27 +46,35 @@ Cpp11BracedListStyle: false
 FixNamespaceComments: true
 NamespaceIndentation: "None"
 PointerAlignment: "Right"
-SortIncludes: true
-SortUsingDeclarations: true
-SpacesInAngles: false
+SortIncludes: "CaseSensitive"
```
[the remark this answers is not in the document]
Comment (Hadatko) [reply .clang-format:49]: Actually i would like to keep it, but restructualize includes in files. If you want keep any include in specific place we just need insert newline before/after that particular case. Value i choosed is based on previous true value. I want do something similar based on my current experience
first header file of declaration
<space>
<erpc includes>
<space>
<nxp includes>
<space>
<freertos includes>
<space>
<other includes>

**Comment by Hadatko** [comment]
Hi @isolansky , if we want to be consistent we can take a look how it looks like in initializing contructor. Newline is before comma. Then we need change both.

image

**Comment by isolansky** [comment]
From my understanding, goal is to have code, easily mergable with other projects (under MCUX) or using MSDK in other way.
From my experiences I always got commas at the line end from OSS either from in-company code. So thus asking if there is any other kind of strong motivation for this. If it is just personal feeling, or some big related project.

**Comment by Hadatko** [comment]
I think you are mixing apples and pears ;) We are trying to align C++ code style with NXP code style. But NXP doesn't have C++ code style (afaik) so we should discuss what is best for C++ specific cases. My suggestion looks like block so easier to read.

Anyway as mentioned before. If we do your requested change it would lead to inconsistency with constructor declaration. So we need keep current settings or change both.

(it looks this type of inconsistency is present in webkit. Do you still like Mozilla? They are using my proposed change :D)

image
image
image

**Comment by Hadatko** [comment]
@MichalPrincNXP @isolansky Your requests are placed. I noticed that clang check was not call on PR change so i added this option and moved excluded files list to its own file so we can have one source of excluded files for two different usage of clang format.

**Comment by isolansky** [comment]
Thank you, also for point out that include sorting based on grouping. This will be subject of future consideration.

**Comment by Hadatko** [comment]
@isolansky That sounds great. Also in that case you would have to use:
IncludeBlocks: Preserve

**Comment by Hadatko** [comment]
@MichalPrincNXP this should be ready too.

**Review by MichalPrincNXP — APPROVED** [review]

**Comment by MichalPrincNXP** [comment]
thank you, Dusan

**Pull request merged**